### PR TITLE
upgrade to TS SDK 0.16.x

### PR DESCRIPTION
**Pull request by vkarpov15**
Took a bit to figure out how to upgrade, but samples-typescript makes it pretty easy.

* createActivityHandle -> proxyActivities from here: https://github.com/temporalio/samples-typescript/blob/main/hello-world/src/workflows.ts
* createWorkflowHandle -> start, must provide `workflowId` from here: https://github.com/temporalio/samples-typescript/blob/main/hello-world/src/execute-workflow.ts
* setListener -> setHandler from here: https://github.com/temporalio/samples-typescript/blob/main/signals-queries/src/workflows.ts

### CODE REVIEW

**Comment by vkarpov15** [comment]
@bergundy @lorensr take a look :+1:

File: src/api/createCart.ts [hunk 1]
```diff
@@ -3,9 +3,11 @@ import { client, taskQueue } from './client';
 import { cartWorkflow } from '../workflows';
 
 export default async function postCart(req: Request, res: Response) {
-  const workflow = client.createWorkflowHandle(cartWorkflow, { taskQueue });
+  const workflowId = 'create-cart-' + Date.now();
```
Comment (lorensr) [inline src/api/createCart.ts:6]: Using 

https://www.npmjs.com/package/uuid

would avoid duplicate workflow ID errors when multiple requests are handled in the same millisecond.

Comment (lorensr) [inline src/api/createCart.ts:6]: Using

https://www.npmjs.com/package/uuid

would avoid duplicate workflow ID errors when multiple requests are handled in the same millisecond.

File: src/test/workflows.test.ts [hunk 2]
```diff
@@ -40,86 +40,82 @@ describe('cart workflow', function() {
...
-  beforeEach(function() {
+  beforeEach(async function() {
     sendStub.resetHistory();
 
     const client = new WorkflowClient();
 
-    workflow = client.createWorkflowHandle(cartWorkflow, { taskQueue });
+    handle = await client.start(cartWorkflow, {
+      taskQueue,
+      workflowId: 'cart-test-' + Date.now(),
```
Comment (lorensr) [inline src/test/workflows.test.ts:50]: Perhaps change this to uuid as well in case really fast or parallelized tests?

**Review by lorensr — APPROVED** [review]

**Pull request merged**